plugins/chat/lib/chat_channel_fetcher.rb

```diff
@@ -43,40 +41,66 @@ def self.generate_allowed_channel_ids_sql(guardian)
           allowed_category_ids: guardian.allowed_category_ids,
         )
         .to_sql
-    }
 
+    dm_channel_sql = ""
+    if !exclude_dm_channels
+      dm_channel_sql = <<~SQL
       UNION
 
       -- secured direct message chat channels
       #{
-      ChatChannel
-        .select(:id)
-        .joins(
-          "INNER JOIN direct_message_channels ON direct_message_channels.id = chat_channels.chatable_id
+        ChatChannel
+          .select(:id)
+          .joins(
+            "INNER JOIN direct_message_channels ON direct_message_channels.id = chat_channels.chatable_id
             AND chat_channels.chatable_type = 'DirectMessage'
         INNER JOIN direct_message_users ON direct_message_users.direct_message_channel_id = direct_message_channels.id",
-        )
-        .where("direct_message_users.user_id = :user_id", user_id: guardian.user.id)
-        .to_sql
-    }
+          )
+          .where("direct_message_users.user_id = :user_id", user_id: guardian.user.id)
+          .to_sql
```

Comment:
Do you think we should add `LIMIT SiteSettingexperimental_hashtag_search_result_limit` . Not sure about that as results are combined from different sources but for sure there is not need to return more results than max capacity

Reply:
Not here because this is purely to say which channel IDs the user is allowed to access, it needs to be all possible channel IDs they can access. The actual limit is applied to `secured_public_channel_search` below